src/netdev.c

```diff
@@ -1018,44 +1018,35 @@ ni_netdev_supports_arp(ni_netdev_t *dev)
 	return FALSE;
 }
 
-static size_t
-__ni_netdev_name_is_valid(const char *ifname)
+const char *
+ni_netdev_name_is_valid(const char *ifname)
 {
-	size_t i, len = ni_string_len(ifname);
+	const char *black_list[] = {
+		"all", "default", NULL
+	}, **ptr;
 
+	size_t i, len = ni_string_len(ifname);
 	if (!len || len >= IFNAMSIZ)
-		return 0;
+		return "Rejecting interface name due to excesive length";
```

Comment:
The suggestion of changing ni_netdev_name_is_valid to return const char * from ni_bool_t and adjusting all calls came from a IRC conversation with @mtomaschewski, I'm not sure if there is another reason for doing it this way (or if I misunderstood something) as just moving the ni_error call as you suggest seems a much more simpler way of achieving the same.

Reply:
I think this was a bad idea. The current patch improves / uses the reason to reject the interface name at exactly 1 place AFAIS (far too less) and causes _many_ changes, because it _inverts_ the (bool evaluated) result of the function.
What about to keep this  function as is and add a ``const char * ni_netdev_name_validate(const char *name);`` function which returns a reason like "too long" usable to format the full (user visible) message ``dbus_set_error(error, DBUS_ERROR_INVALID_ARGS, "invalid interface %s: %s", ifname, reason)``.
Then, ``ni_netdev_name_is_valid`` can  use the ``_validate()`` function inside.
And please get rid of the underscores at the begin of the inner ``__ni_netdev_name_is_valid``.

Reply:
I'd tend to use a new function which returns a reason, for example:
```
const char *
ni_netdev_name_validate(const char *ifname)
{
        /* reserved names, e.g. used in sysctls */
        const char *black_list[] = {
                "all", "default", NULL
        }, **ptr;
        size_t len, i;

        len  = ni_string_len(ifname);
        if (len == 0)
                return "empty";

        if (len >= IFNAMSIZ)
                return "too long";

        if (!isalnum((unsigned char)ifname[0]))
                return "1st character not alphanumeric";

        for(i = 0; i < len; ++i) {
                if(isalnum((unsigned char)ifname[i]) ||
                                ifname[i] == '-' ||
                                ifname[i] == '_' ||
                                ifname[i] == '.')
                        return "disallowed characters";
        }

        for (ptr = black_list; *ptr; ptr++) {
                if (ni_string_eq(*ptr, ifname))
                        return "reserved name";
        }

        return NULL;
}

ni_bool_t
ni_netdev_name_is_valid(const char *ifname)
{
        return ni_netdev_name_validate(ifname) == NULL;
}
```

Reply:
Based on the feedback received I'd like to close the PR and reopen in the future with a different approach.